sdk/storage/src/blobs/block_blob_client.cpp

```diff
@@ -75,6 +77,64 @@ namespace Azure { namespace Storage { namespace Blobs {
         options.Context, *m_pipeline, m_blobUrl.ToString(), content, protocolLayerOptions);
   }
 
+  BlobContentInfo BlockBlobClient::UploadFromBuffer(
+      const uint8_t* buffer,
+      std::size_t bufferSize,
+      const UploadBlobOptions& options) const
+  {
+    constexpr int64_t c_defaultBlockSize = 8 * 1024 * 1024;
+    constexpr int64_t c_maximumNumberBlocks = 50000;
+    constexpr int64_t c_grainSize = 4 * 1024;
```

Comment:
`c_grainSize ` should also be optional parameter. If HTTP stack decides to read some other value than 4kb per reading, we can optimize with changing this value.
However, not a priority. Only need to doc this if you don't want to change it.

Reply:
This is not for HTTP stack optimization or something. It's designed for two purposes

1. For page blob, this value must be a multiple of 512.
2. For other kinds of blob, like block blob, I just feel it's kind of strange if the block size is not aligned to some power-of-2 boundaries. For example, uploading a 700KB blob with 6 threads, 120KB block size looks better than 119467 bytes.

Reply:
And necessity of making it customizable by users?

Reply:
Customer might chose their own HTTP stack, the grain size can be used to align with the HTTP stack's reading size per read for a better performance.

Reply:
I didn't see how it's relevant to performance.